Shopify/cli
# Simplify code around migrating extensions

### WHY are these changes introduced?
As we added more and more modules to be migrated, we kept adding duplicated logic. 



### WHAT is this pull request doing?
- Unify the logic to "Get modules to migrate by type", create a generic function and call it with different module type-mappings
- Remove duplicated logic and tests.



### How to test your changes?
Try to import extensions and then migrate them, it should work as always
Example:
- Create a flow action extension in your partner dashboard
- Run `shopify app import-extension` and import the extension you just created
- Run `shopify app deploy` and it should ask for confirmation to migrate that extension.


### Post-release steps



### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
